tox.ini

```diff
@@ -12,47 +12,60 @@ skip_missing_interpreters = true
 
 [testenv]
 description = run the tests with pytest under {basepython}
-setenv = COVERAGE_FILE={toxworkdir}/.coverage.{envname}
+setenv = VIRTUALENV_DOWNLOAD = 0
+         PIP_DISABLE_VERSION_CHECK = 1
+         COVERAGE_FILE = {toxworkdir}/.coverage.{envname}
 passenv = http_proxy https_proxy no_proxy SSL_CERT_FILE TOXENV CI TRAVIS TRAVIS_* APPVEYOR APPVEYOR_* CODECOV_*
+deps = setuptools >= 29.0.0
+       setuptools_scm >= 2.0.0, < 3
+       wheel >= 0.31.0
 extras = testing
-commands = pytest {posargs:--cov-config="{toxinidir}/tox.ini" --cov="{envsitepackagesdir}/tox" --timeout=180 tests}
+changedir = {toxinidir}/tests
+commands = pytest {posargs:--cov="{envsitepackagesdir}/tox" --cov-config="{toxinidir}/tox.ini"  --timeout=180 .}
 
 [testenv:docs]
 description = invoke sphinx-build to build the HTML docs and check that all links are valid
 whitelist_externals = sphinx-build
 basepython = python3.6
 extras = docs
+changedir = {toxinidir}
 commands = sphinx-build -d "{toxworkdir}/docs_doctree" doc "{toxworkdir}/docs_out" --color -W -bhtml
 
 
 [testenv:fix-lint]
+description = run static analysis and style check using flake8
 basepython = python3.6
 passenv = {[testenv]passenv}
           HOMEPATH
           # without PROGRAMDATA cloning using git for Windows will fail with an
           # `error setting certificate verify locations` error
           PROGRAMDATA
 extras = lint
-description = run static analysis and style check using flake8
+changedir = {toxinidir}
```

Comment:
heh, `pre-comit` will `cd` right back to the repository root, not sure this is useful